src/mork/api/routes/tasks.py

```diff
@@ -2,8 +2,40 @@
 
 import logging
 
-from fastapi import APIRouter
+from celery.result import AsyncResult
+from fastapi import APIRouter, Depends, Response, status
+
+from mork.api.auth import authenticate_api_key
+from mork.api.models import TaskCreate, TaskResponse, TaskStatus, TaskType
+from mork.worker.tasks import TASK_TYPE_TO_FUNC
 
 logger = logging.getLogger(__name__)
 
-router = APIRouter(prefix="/tasks")
+router = APIRouter(prefix="/tasks", dependencies=[Depends(authenticate_api_key)])
+
+
+@router.post("/", status_code=status.HTTP_202_ACCEPTED)
+async def create_task(task_in: TaskCreate, response: Response) -> TaskResponse:
+    """Create a new task."""
+    celery_task = TASK_TYPE_TO_FUNC[task_in.type]
+    result = celery_task.delay()
+
+    task = TaskResponse(id=result.task_id, status=TaskStatus.PENDING)
+    response.headers["location"] = router.url_path_for(
+        "get_task_status", **{"task_id": task.id}
+    )
+    return task
+
+
+@router.get("/")
+async def get_available_tasks() -> dict:
```

Comment:
This is confusing. I expect to have all processing task in celery. IMO this should be an `OPTIONS` on the endpoint `/tasks`